server/src/main/java/org/opensearch/rest/extensions/RestSendToExtensionAction.java

```diff
@@ -219,6 +219,10 @@ public void handleResponse(RestExecuteOnExtensionResponse response) {
             @Override
             public void handleException(TransportException exp) {
                 logger.debug("REST request failed", exp);
+                // On failure the original request params and content aren't consumed
+                // which gives misleading error messages, so we just consume them here
+                request.params().keySet().stream().forEach(p -> request.param(p));
+                request.content();
```

Comment:
Do we want to add a test for this bug?

Reply:
This is in a transport response handler, so would best be handled by an IT triggering a test REST request that times out.  Don't think we can cover that in a unit test, and the IT "coverage" would be on SDK side.

Reply:
Let's create an issue to handle this or if we have one, let's link it here. Don't want to hold the PR.

Reply:
https://github.com/opensearch-project/opensearch-sdk-java/issues/826

Reply:
@owaiskazi19 @dbwiddis Can this conversation be resolved?  PR can't be merged until that is done